nerfstudio/scripts/datasets/process_project_aria.py

```diff
@@ -0,0 +1,227 @@
+    assert nearest_pose_idx != -1, f"Could not find pose for {capture_time_ns}"
+    t_world_device = t_world_devices.t_world_devices[nearest_pose_idx]
+
+    # Compute the world to camera transform.
+    t_world_camera = t_world_device @ camera_calibration.t_device_camera @ T_ARIA_NERFSTUDIO
+
+    return AriaImageFrame(
+        camera=camera_calibration,
+        file_path=file_path,
+        t_world_camera=t_world_camera,
+        timestamp_ns=capture_time_ns,
+    )
+
+
+def to_nerfstudio_frame(frame: AriaImageFrame) -> Dict:
+    return {
+        "fl_x": frame.camera.fx,
+        "fl_y": frame.camera.fy,
+        "cx": frame.camera.cx,
+        "cy": frame.camera.cy,
+        "distortion_params": frame.camera.distortion_params.tolist(),
+        "w": frame.camera.width,
+        "h": frame.camera.height,
+        "file_path": frame.file_path,
+        "transform_matrix": frame.t_world_camera.to_matrix().tolist(),
+        "timestamp": frame.timestamp_ns,
+    }
+
+
+@dataclass
+class ProcessProjectAria:
+    """Processes Project Aria data i.e. a VRS of the raw recording streams and the MPS attachments
+    that provide poses, calibration, and 3d points. More information on MPS data can be found at:
+      https://facebookresearch.github.io/projectaria_tools/docs/ARK/mps.
+    """
+
+    vrs_file: Path
+    """Path to the VRS file."""
+    mps_data_dir: Path
+    """Path to Project Aria Machine Perception Services (MPS) attachments."""
+    output_dir: Path
+    """Path to the output directory."""
+
+    def main(self) -> None:
+        """Generate a nerfstudio dataset from ProjectAria data (VRS) and MPS attachments."""
+        # Create output directory if it doesn't exist.
+        self.output_dir.mkdir(parents=True, exist_ok=True)
+
+        provider = create_vrs_data_provider(self.vrs_file.absolute().as_posix())
+        assert provider is not None, "Cannot open file"
+
+        name_to_camera = get_camera_calibs(provider)
+
+        print("Getting poses from closed loop trajectory CSV...")
+        trajectory_csv = self.mps_data_dir / "closed_loop_trajectory.csv"
+        t_world_devices = read_trajectory_csv_to_dict(trajectory_csv.absolute().as_posix())
+
+        name = "camera-rgb"
+        stream_id = provider.get_stream_id_from_label(name)
+
+        # create an AriaImageFrame for each image in the VRS.
+        print("Creating Aria frames...")
+        aria_frames = [
+            to_aria_image_frame(provider, index, name_to_camera, t_world_devices, self.output_dir)
+            for index in range(0, provider.get_num_data(stream_id))
+        ]
+
+        # create the NerfStudio frames from the AriaImageFrames.
+        print("Creating NerfStudio frames...")
+        RGB_VALID_RADIUS = 707.5
+        nerfstudio_frames = {
+            "camera_model": ARIA_CAMERA_MODEL,
+            "frames": [to_nerfstudio_frame(frame) for frame in aria_frames],
+            "fisheye_crop_radius": RGB_VALID_RADIUS,
+        }
+
```

Comment:
It could also be helpful to save the point cloud here, so #2521 can use it for initialization (once it's merged):

```python
        # import pandas as pd
        # import open3d as o3d
        points_path = self.mps_data_dir / "global_points.csv.gz"
        if points_path.exists():
            print("Found global points, saving to CSV...")
            df = pd.read_csv(points_path, compression="gzip")
            points = df[["px_world", "py_world", "pz_world"]].values
            pcd = o3d.geometry.PointCloud()
            pcd.points = o3d.utility.Vector3dVector(points)
            ply_file_path = self.output_dir / "global_points.ply"
            o3d.io.write_point_cloud(str(ply_file_path), pcd)

            nerfstudio_frames["ply_file_path"] = "global_points.ply"
        else:
            print("No global points found!")
```

We would also need to add support for Fisheye624 in the splatting pipeline... ideally we would integrate the camera model directly into gsplat/the rasterization kernels, but maybe just undistorting the label images is fine too? Seems like that's how the OpenCV fisheye model is being handled right now.

cc @kerrj, @vye16

Reply:
Mind if I leave this for a future commit? Since the main nerf pipeline won't use the points (right?)